### DISCUSSION TITLE
Hiding footer and navigation on auto-generated docs

### DISCUSSION THREAD

**Discussion by astrochun** [mkdocstrings/mkdocstrings#461] [Q&A] [answered]
Hi, I looked through the docs and couldn't find a way to do this in a manner consistent with `material`.

In short, Material for mkdocs allow the ability to hide navigation and footer by including the following at the top of the markdown file:
```md
---
hide:
  - footer
  - navigation
---
```

I've followed the [recipe for auto-generated markdown](https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages) for modules, etc. I'm wondering if there's a way for the above to also occur for these auto-generated pages.

Unclear if this is a new feature or if there's a configuration hack

**Comment by oprypin** [comment, marked as answer]
For that recipe, all you need to do is print that exact thing into the file  prior to printing the `:::` directive.

```python
header = '''\
---
hide:
  - footer
  - navigation
---'''
print(header, file=fd)
```

**Reply by astrochun** [reply]
I see, so you mean modify the `gen_ref_pages` module. I'll give that a try. Thanks for the quick response.

**Reply by astrochun** [reply]
This did the trick!
